### ISSUE TITLE
Is it your idea to add batch normalization and replace upconv with bilinear upsample?

### ISSUE THREAD

**Issue by fanfanda** [milesial/Pytorch-UNet#55]

**Comment by milesial** [comment]
Hi, yes, the bilinear upsample is a way to have the model use less memory, even if the results are not as good. You can use upconv with `bilinear=False` in the net construction.
About the batch norm, I believe there was some discussion on the Kaggle forums and kernels about it, so I decided to add it.

**Comment by fanfanda** [comment]
Thanks for your reply. In my recent paper experiment, I retained two of your changes. Do I need to cite your project? I hope you can contact me by email. My email is (fanfanda@ict.ac.cn). Thank you again.

**Comment by milesial** [comment]
If you wish to cite me you can link the project URL, but I have to tell you that these changes are not backed by any theoretical study, I did it mainly because my GPU did not have enough memory :)
Make sure to follow the GNU GPL v3 license.

**Comment by fanfanda** [comment]
ok, thanks~
